src/librustc_borrowck/borrowck/mod.rs

```diff
@@ -1213,7 +1213,7 @@ impl<'a, 'tcx> BorrowckCtxt<'a, 'tcx> {
                             db.span_label(
                                 let_span,
                                 format!("consider changing this to `{}`",
-                                         snippet.replace("ref ", "ref mut "))
+                                         snippet.replacen("ref ", "ref mut ", 1))
```

Comment:
This is (a) not requiring the `ref` to be at the start of the span (this might be problematic) and (b) only matching the first space after it (this is okay if we want to preserve weird formatting). This might be fine for all inputs we consider now, but matching `^ref ` seems safer in general.

Reply:
thank you. I tried to find a way to match the whole word at first, but failed, since the `core::str::Pattern` used by `String::replace` is real naive nowadays(or I'm naive, could you give me some tips?), so I couldn't restrict a single word without extern crate like regex.

I think the raw replacement here is a bit untidy comparing to the common diagnostics that lint changes with *certain* `Span` in rustc. I have a idea that get the `ref` span from `let_span` indirectly(`BindingAnnotation` has no span), thus:
```diff
 let ref my_ref @ _ = 0;
-    ^^^^^^^^^^^^^^ = help
+    ^^^ = help
```
How about this?